### ISSUE TITLE
Relax version of typing-extensions

### ISSUE THREAD

**Issue by SonOfLilit** [pydantic/pydantic#8818]
### Initial Checks

- [X] I have searched GitHub for a duplicate issue and I'm sure this is something new
- [X] I have searched Google & StackOverflow for a solution and couldn't find anything
- [X] I have read and followed [the docs](https://docs.pydantic.dev) and still think this is a bug
- [X] I am confident that the issue is with pydantic (not my code, or another library in the ecosystem like [FastAPI](https://fastapi.tiangolo.com) or [mypy](https://mypy.readthedocs.io/en/stable))

### Description

`pydantic==1.10.4` depends on `typing-extensions==4.3.0`. This e.g. makes it harder to run it side by side with (a renamed) `pydantic v2` (which you'd want to do to switch gradually).

Since `typing-extensions` promise to follow semantic versioning, and since all tests pass with `typing-extensions==4.6.0`, I propose relaxing the requirement to be `>=4.3.0,<5` as recommended in their docs.

### Example Code

```Python
typing-extensions>=4.3.0,<5
```

### Python, Pydantic & OS Version

```Text
pydantic version: 1.10.14
            pydantic compiled: False
                 install path: /Users/aursaraf/dev/pydantic/pydantic
               python version: 3.10.6 (v3.10.6:9c7b4bd164, Aug  1 2022, 17:13:48) [Clang 13.0.0 (clang-1300.0.29.30)]
                     platform: macOS-13.2.1-arm64-arm-64bit
     optional deps. installed: ['devtools', 'dotenv', 'email-validator', 'typing-extensions']
```

### Affected Components

- [X] [Compatibility between releases](https://docs.pydantic.dev/changelog/)
- [ ] [Data validation/parsing](https://docs.pydantic.dev/concepts/models/#basic-model-usage)
- [ ] [Data serialization](https://docs.pydantic.dev/concepts/serialization/) - `.model_dump()` and `.model_dump_json()`
- [ ] [JSON Schema](https://docs.pydantic.dev/concepts/json_schema/)
- [ ] [Dataclasses](https://docs.pydantic.dev/concepts/dataclasses/)
- [ ] [Model Config](https://docs.pydantic.dev/concepts/config/)
- [ ] [Field Types](https://docs.pydantic.dev/api/types/) - adding or changing a particular data type
- [ ] [Function validation decorator](https://docs.pydantic.dev/concepts/validation_decorator/)
- [ ] [Generic Models](https://docs.pydantic.dev/concepts/models/#generic-models)
- [ ] [Other Model behaviour](https://docs.pydantic.dev/concepts/models/) - `model_construct()`, pickling, private attributes, ORM mode
- [ ] [Plugins](https://docs.pydantic.dev/) and integration with other tools - mypy, FastAPI, python-devtools, Hypothesis, VS Code, PyCharm, etc.

**Comment by SonOfLilit** [comment]
https://typing-extensions.readthedocs.io/en/latest/#versioning-and-backwards-compatibility

**Comment by sydney-runkle** [comment]
Closed by #8819
